bookkeeper-server/src/main/java/org/apache/bookkeeper/util/ZkUtils.java

```diff
@@ -240,7 +242,12 @@ public void operationComplete(int rc, List<String> ledgers) {
 
         synchronized (ctx) {
             while (!ctx.done) {
-                ctx.wait();
+                try {
+                    ctx.wait(TimeUnit.SECONDS.toMillis(OP_TIME_OUT_SEC));
```

Comment:
In this case, if no one calls `ctx.done = true`,  the thread will just keep waiting in 2sec blocks, but it will never exit, right?

Reply:
umm.. if no one calls `ctx.done = true; ctx.notifyAll();` at line#238 then `ctx.wait(timeout)` will throw `InterruptedException` and `ctx.done will be true;` at line#249 which takes out thread out of while loop. So, it will not keep waiting but exit. right?